packages/react-native/src/Authenticator/hooks/useFieldValues/useFieldValues.ts

```diff
@@ -112,5 +132,10 @@ export default function useFieldValues<FieldType extends TypedField>({
     handleSubmit?.(submitValue);
   };
 
-  return { fields: fieldsWithHandlers, disableFormSubmit, handleFormSubmit };
+  return {
+    fields: fieldsWithHandlers,
+    disableFormSubmit,
+    fieldValidationErrors: { ...fieldValidationErrors, ...validationErrors },
```

Comment:
Definitely may be missing something here, but curious if `validationErrors` still needs to be merged with `fieldValidationErrors` since (if I am reading correctly} `runValidations` has already concatenated the values contained in `validationErrors` [here](https://github.com/aws-amplify/amplify-ui/pull/4107/files#diff-52e68fc0024cea567f183a7a72a8813c90a961a5ee667f03abc479015cf676baR211)?

Reply:
This covers the case where there is no blur or change event but there are state machine errors. We don't want to lose them.

Reply:
Maybe we should move the logic related to the machine errors out of `runValidations` and create a util to be used here so we can have one source of truth in how state machine validations are merged with the local state validations?

Reply:
I was hesitant to make extra changes to this bc we're removing it with the upcoming auth work, what do you think, is it a blocker?

Reply:
My concern is mostly just that if for some reason the merging related code needs to get updated, it might be difficult to understand what's happening here without prior context. That being said if anything does come up it'll most likely be addressed by you or i so okay with it as is